lib/elements/satin_column.py

```diff
@@ -696,19 +770,23 @@ def walk(self, path, start_pos, start_index, distance):
                 distance_remaining -= segment_length
                 pos = segment_end
 
-    def plot_points_on_rails(self, spacing, offset_px=(0, 0), offset_proportional=(0, 0)):
+    def plot_points_on_rails(self, spacing, offset_px=(0, 0), offset_proportional=(0, 0), use_random=False
+                             ) -> typing.List[typing.Tuple[Point, Point]]:
         # Take a section from each rail in turn, and plot out an equal number
         # of points on both rails.  Return the points plotted. The points will
         # be contracted or expanded by offset using self.offset_points().
 
-        def add_pair(pos0, pos1):
-            pos0, pos1 = self.offset_points(pos0, pos1, offset_px, offset_proportional)
-            points[0].append(pos0)
-            points[1].append(pos1)
+        # pre-cache ramdomised parameters to avoid property calls in loop
+        if use_random:
+            seed = prng.joinArgs(self.random_seed, "satin-points")
+            offset_proportional_min = np.array(offset_proportional) - self.random_width_decrease
+            offset_range = (self.random_width_increase + self.random_width_decrease)
+            spacing_range = spacing * self.random_zigzag_spacing / 100
```

Comment:
If I am not wrong you don't need to divide through 100 anymore (already done in line 120)

Reply:
Good catch, fixed